tensorflow_lite_support/python/task/text/nl_classifier.py

```diff
@@ -0,0 +1,163 @@
+
+  def __init__(self, options: NLClassifierOptions,
+               cpp_classifier: _CppNLClassifier) -> None:
+    """Initializes the `NLClassifier` object."""
+    # Creates the object of C++ NLClassifier class.
+    self._options = options
+    self._classifier = cpp_classifier
+
+  @classmethod
+  def create_from_file(cls, file_path: str) -> "NLClassifier":
+    """Creates the `NLClassifier` object from a TensorFlow Lite model.
+
+    Args:
+      file_path: Path to the model.
+
+    Returns:
+      `NLClassifier` object that's created from the model file.
+    Raises:
+      ValueError: If failed to create `NLClassifier` object from the provided
+        file such as invalid file.
+      RuntimeError: If other types of error occurred.
+    """
+    base_options = _BaseOptions(file_name=file_path)
+    options = NLClassifierOptions(base_options=base_options)
+    return cls.create_from_options(options)
+
+  @classmethod
+  def create_from_options(cls, options: NLClassifierOptions) -> "NLClassifier":
+    """Creates the `NLClassifier` object from NL classifier options.
+
+    Args:
+      options: Options for the NL classifier task.
+
+    Returns:
+      `NLClassifier` object that's created from `options`.
+    Raises:
+      ValueError: If failed to create `NLClassifier` object from
+        `NLClassifierOptions` such as missing the model or if any of the
+        classification options is invalid.
+      RuntimeError: If other types of error occurred.
+    """
+    classification_options = options.classification_options
+
+    if classification_options.max_results == 0:
+      raise ValueError("Invalid `max_results` option: value must be != 0")
+
+    if classification_options.category_name_allowlist is not None and \
+        classification_options.category_name_denylist is not None:
+      if len(classification_options.category_name_allowlist) > 0 and \
+          len(classification_options.category_name_denylist) > 0:
+        raise ValueError(
+          "`category_name_allowlist` and `category_name_denylist` are mutually "
+          "exclusive options.")
+
+    classifier = _CppNLClassifier.create_from_options(
+        options.base_options.to_pb2())
+    return cls(options, classifier)
+
+  def classify(self, text: str) -> classifications_pb2.ClassificationResult:
+    """Performs actual NL classification on the provided text.
+
+    Args:
+      text: the input text, used to extract the feature vectors.
+
+    Returns:
+      classification result.
+
+    Raises:
+      ValueError: If any of the input arguments is invalid.
+      RuntimeError: If failed to perform NL classification.
+    """
+    classification_result = self._classifier.classify(text)
+    classification_result = classifications_pb2.ClassificationResult.\
+        create_from_pb2(classification_result)
+    return self._postprocess(classification_result)
+
+  def _postprocess(self, result: classifications_pb2.ClassificationResult):
```

Comment:
Ideally, this logic should be checked into the C++ layer (in fact in C++, we can reuse the existing classification_postprocessor), and be shared across all platforms. My concern is the C++/Java NLClassifier don't support those options (i.e. allowlist/denylist) at this point, and this gonna be a little confusing why we only support it in Python. I'd be OK if we completely ignore this feature at this point. Otherwise, I would enable it from C++ and expose the options in other languages.

Reply:
SGTM. Let me update the internal code to ignore the ClassificationOptions entirely for now. We'll look into implementing it in the C++ layer later.